rust/ql/src/queries/unusedentities/UnusedVariable.qll

```diff
@@ -24,6 +38,9 @@ predicate isAllowableUnused(Variable v) {
   // in a macro expansion
   v.getPat().isInMacroExpansion()
   or
+  // declared in an incomplete callable
+  v.getEnclosingCfgScope() instanceof IncompleteCallable
```

Comment:
While this works, it would perhaps be more correct to only exclude results when there is an unexpandable macro in the _scope_ that the variable is defined. However, I realize that the concept of variable scope is not currently exposed, so perhaps that is better done follow-up.

Reply:
What is required to expose the concept of a variable scope?

I've added a test that exposes the situation I think we're talking about here.

Reply:
We would have to expose the `VariableScope` class from `VariableImpl.qll`.

Reply:
OK, I've added it to the existing improvements issue for `rust/unused-variable`.  It sounds like a straightforward follow-up, apart from the need to go through another cycle of DCA / testing.

Is there any reason `VariableScope` *shouldn't* be exposed publically?

Reply:
> Is there any reason `VariableScope` _shouldn't_ be exposed publically?

Only that we haven't needed it before. We may also have to change it slightly, for example right now `x` and `y` are considered to be in the same scope in
```rust
let x = ...;
let y = ...;
```